source/extensions/common/dynamic_forward_proxy/dns_cache_impl.cc

```diff
@@ -408,6 +408,7 @@ void DnsCacheImpl::finishResolve(const std::string& host,
     primary_host_info->host_info_->setAddresses(new_address, std::move(address_list));
 
     runAddUpdateCallbacks(host, primary_host_info->host_info_);
+    primary_host_info->host_info_->setFirstResolveComplete();
```

Comment:
do you mind runtime guarding this just out of abject paranoia?

Reply:
ACK

Reply:
done